scripts/toyzero/train_test_split

```diff
@@ -0,0 +1,206 @@
+        dest    = 'test_size',
+        help    = (
+            'size of the test dataset. '
+            'If test-size <= 1, then it is interpreted as a fraction'
+        ),
+        type    = float,
+    )
+
+    return parser.parse_args()
+
+class CopyingWorker:
+    # pylint: disable=missing-class-docstring
+
+    def __init__(self, root, outdir_real, outdir_fake):
+        self._root        = root
+        self._outdir_real = outdir_real
+        self._outdir_fake = outdir_fake
+
+    def __call__(self, index_fname):
+        (index, fname) = index_fname
+
+        path_real = os.path.join(self._root, DIR_REAL, fname)
+        path_fake = os.path.join(self._root, DIR_FAKE, fname)
+
+        fname_out = f'sample_{index}.npz'
+
+        shutil.copy(path_real, os.path.join(self._outdir_real, fname_out))
+        shutil.copy(path_fake, os.path.join(self._outdir_fake, fname_out))
+
+        return (index, fname, fname_out)
+
+def create_config_from_cmdargs(cmdargs):
+    return Config(
+        root      = cmdargs.root,
+        outdir    = cmdargs.outdir,
+        seed      = cmdargs.seed,
+        shuffle   = cmdargs.shuffle,
+        test_size = cmdargs.test_size,
+        val_size  = cmdargs.val_size,
+    )
+
+def prepare_image_lists(config):
+    images = collect_toyzero_images(config.root)
+    prg    = np.random.default_rng(config.seed)
+
+    train_indices, val_indices, test_indices = train_val_test_split(
+        len(images), config.val_size, config.test_size, config.shuffle, prg
+    )
+
+    train_images = [ images[i] for i in train_indices ]
+    val_images   = [ images[i] for i in val_indices   ]
+    test_images  = [ images[i] for i in test_indices  ]
+
+    return (train_images, val_images, test_images)
+
+def extract_images(config, split, images, rename_map):
+    outdir_split = os.path.join(config.outdir,  split)
+    outdir_real  = os.path.join(outdir_split,   DIR_REAL)
+    outdir_fake  = os.path.join(outdir_split,   DIR_FAKE)
+
+    os.makedirs(outdir_real, exist_ok = True)
+    os.makedirs(outdir_fake, exist_ok = True)
+
+    progbar = tqdm.tqdm(
+        desc = 'Splitting', total = len(images), dynamic_ncols = True
+    )
+
+    worker = CopyingWorker(config.root, outdir_real, outdir_fake)
+
+    with multiprocessing.Pool() as pool:
+        for (index, fname_src, fname_dst) in pool.imap_unordered(
+            worker, enumerate(images)
+        ):
+            rename_map.append((split, index, fname_dst, fname_src))
+            progbar.update()
+
+    progbar.close()
+
+def save_config(config):
+    config_path = os.path.join(config.outdir, 'config.json')
+    config_dict = {
+        'script'    : 'toyzero/train_test_split',
+        'seed'      : config.seed,
+        'shuffle'   : config.shuffle,
+        'test_size' : config.test_size,
+        'val_size'  : config.val_size,
+    }
+
+    with open(config_path, 'wt') as f:
+        json.dump(config_dict, f, indent = 4)
+
+def save_map(config, rename_map):
```

Comment:
@usert5432  maybe add a comment here to indicate the direction of the map. something like "map original name to new file names with contiguous indices".

Reply:
Thank you for the review @YHRen!
> @usert5432 maybe add a comment here to indicate the direction of the map. something like "map original name to new file names with contiguous indices".

I have thought about it, but I decided not add a comment, since just a few lines below, a header is defined for this map:
```python
df = pd.DataFrame(rename_map, columns = [ 'split', 'index', 'fname_dest', 'fname_src' ])
```
So, maybe it is obvious what it means. Also while I have called it a `map` it is not a map, but a list. Do you think it is ok to leave it as is?

Reply:
@usert5432  Looks good. Feel free to merge.

Reply:
Ok, Thanks!